source/declarations.tex

```diff
@@ -4239,3 +4240,414 @@
+  return x;
+}
+\end{codeblock}
+\end{example}
+
+\pnum
+If the \grammarterm{contract-level}
+of a \grammarterm{contract-attribute-specifier} is absent,
+it is assumed to be \tcode{default}.
+\begin{note}
+A \tcode{default} \grammarterm{contract-level} is expected to be used
+for those contracts where
+the cost of run-time checking is assumed to be
+small (or at least not expensive) compared to
+the cost of executing the function.
+An \tcode{audit} \grammarterm{contract-level} is expected to be used
+for those contracts where
+the cost of run-time checking is assumed to be
+large (or at least significant) compared to
+the cost of executing the function.
+An \tcode{axiom} \grammarterm{contract-level} is expected to be used
+for those contracts that are formal comments
+and are not evaluated at run-time.
+\end{note}
+
+\pnum
+Multiple contract conditions may be applied to a function type
+with the same or different \grammarterm{contract-level}{s}.
+\begin{example}
+\begin{codeblock}
+int z;
+
+bool is_prime(int k);
+
+void f(int x)
+  [[expects: x > 0]]
+  [[expects audit: is_prime(x)]]
+  [[ensures: z > 10]]
+{
+  @\commentellip@
+}
+\end{codeblock}
+\end{example}
+
+\pnum
+A translation may be performed
+with one of the following \defnx{build levels}{build level}:
+\term{off}, \term{default}, or \term{audit}.
+A translation with build level set to \term{default}
+performs checking for \tcode{default} contracts.
+A translation with build level set to \term{audit}
+performs checking for \tcode{default} and \tcode{audit} contracts.
+If no build level is explicitly selected, the build level is \term{default}.
+The mechanism for selecting the build level is
+\impldef{mechanism for selecting the build level}.
+The translation of a program consisting of translation units
+where the build level is not the same in all translation units
+is conditionally-supported.
+There should be no programmatic way of setting, modifying, or querying
+the build level of a translation unit.
+
+\pnum
+It is unspecified whether a contract
+that would not be checked under the current build level is evaluated.
+If it would evaluate to \tcode{false}, the behavior is undefined.
+During constant expression evaluation\iref{expr.const},
+contract evaluation is performed only for those contracts that are checked.
+
+\pnum
+The \defn{violation handler} of a program is a function of type
+``\opt{\tcode{noexcept}} function of
+(lvalue reference to \tcode{const std::contract_violation})
+returning \tcode{void}'' specified in an
+\impldef{specification of violation handler} manner.
+The violation handler is invoked
+when the predicate of a checked contract evaluates to \tcode{false}
+(called a \defn{contract violation}).
+There should be no programmatic way of
+setting or modifying the violation handler.
+It is \impldef{argument for violation handler}
+how the violation handler is established for a program and
+how the \tcode{std::contract_violation} argument value is set.
```

Comment:
I think we need an "except as specified below" here, since we're about to give some rules that dictate how parts of the `contract_violation` argument value are set.

Reply:
Agreed; also added a cross-reference to [support.contract.cviol].